### PR TITLE
support multipe usage/usage-pairs on MacOS

### PR DESCRIPTION
**Pull request by fengj2006**
This change supports one device has multiple usage/usage-page pairs on MacOS, which is already supported on Windows.

### CODE REVIEW

File: hidapi/hidapi.h [hunk 1]
```diff
@@ -80,7 +80,7 @@ extern "C" {
 
 			/** Pointer to the next device */
 			struct hid_device_info *next;
-		};
+		} HID_DEVICE_INFO;
```
Comment (Youw) [inline hidapi/hidapi.h]: this change is not related to what PR is really about - support usage pages on macOS
it also changes public API, and there is no real need for it, please undo

Comment (fengj2006) [reply hidapi/hidapi.h]: Fixed

**Comment by Youw** [comment]
Fixes #116

File: mac/hid.c [hunk 2]
```diff
@@ -392,6 +406,117 @@ static void process_pending_events(void) {
...
+			}
+			else {
+				cur->next = next;
+			}
+			cur = next;
+		}
+	}
+
+	if (root == NULL) {
+		/* error when generating or parsing usage pairs */
```
Comment (Youw) [inline mac/hid.c]:
```suggestion
		/* error when generating or parsing usage pairs, or device/driver doesn't report usage pairs (correctly) */
```

**Comment by Youw** [comment]
I don't think there is a way to submit your changes to this particular PR since you'd need to commit to author's PR, and you don't have write access to his copy of the repo.

But you can:
- clone original repo (origin);
- add git remote to fengj2006;
- checkout this PR's branch (this andbthevprevious just so you have )
